### PR TITLE
implement utp transport

### PR DESCRIPTION
**Pull request by whyrusleeping**
License: MIT
Signed-off-by: Jeromy <jeromyj@gmail.com>

### CODE REVIEW

File: p2p/net/transport/utp.go [hunk 1]
```diff
@@ -0,0 +1,148 @@
+package transport
```
Comment (jbenet) [inline p2p/net/transport/utp.go]: i thought libp2p got moved out + vendored? i guess not yet?

**Comment by jbenet** [comment]
very nice PR :heart: :+1: 

I would really like to have really harsh tests being run + producing outputs + graphs. we should look into setting this up after 0.4.0 is out.

**Comment by jbenet** [comment]
this LGTM -- RFM?

**Comment by whyrusleeping** [comment]
this is RFM.

libp2p isnt ripped out yet, but i'll get to it soon!

**Comment by jbenet** [comment]
Boom! :)

**Pull request merged**